File: server/models/itemModel.js
const mongoose = require("mongoose");
const Schema = mongoose.Schema;

const itemSchema = new Schema({
  item_id: { type: String, required: true },
  name: { type: String, required: true },
  quantity: { type: Number, required: true },
  category: { type: String, required: true },
  price: { type: Number, required: true },
  status: { type: String, enum: ["in_stock", "out_of_stock"] },
  godown_id:{ type: String, required: true },
  brand: { type: String, required: true },
  attributes: { type: Map, of: String },
  image_url: { type: String, required: true },
});

const Item = mongoose.model("Item", itemSchema);
module.exports = Item;

File: backend/models/itemModel.py
from pydantic import BaseModel
from enum import Enum


class ItemStatus(str, Enum):
    in_stock = "in_stock"
    out_of_stock = "out_of_stock"


class Item(BaseModel):
    item_id: str
    name: str
    quantity: int
    category: str
    price: float
    status: ItemStatus
    godown_id: str
    brand: str
    attributes: dict
    image_url: str
